project4/name_server/tasks/nameserver_task.cpp

```diff
@@ -4,18 +4,16 @@
 
 namespace nameserver::tasks {
 
-/// TODO Instantiate name server
-NameserverTask::NameserverTask(std::shared_ptr<nameserver::Nameserver> nameserver)
+NameserverTask::NameserverTask(
+    std::shared_ptr<nameserver::Nameserver> nameserver)
     : nameserver_(nameserver) {}
 
-/// TODO Set up socket listening
-thread_pool::Task::Status NameserverTask::SetUp() {
-  return Task::SetUp();
-}
-
-/// TODO Listen to all requests. Pass them accordingly to HandleResponse
-/// If there is a command in cmd_queue_, parse and execute.
 thread_pool::Task::Status NameserverTask::RunAtomic() {
+  message_passing::Endpoint endpoint;
+  consistent_hash_msgs::NameServerMessage ns_msg;
+  if (nameserver_->server_->Receive(kTimeout, &ns_msg, &endpoint)) {
```

Comment:
From a design perspective, having the `NameserverTask` directly access the `server_` member is pretty tight coupling and would generally be frowned upon. Why not add a `ReceiveAndHandleRequest()` method to the `Nameserver` class that does exactly that? Otherwise, the `Server` should be a member of the `NameserverTask` class, IMO.

Reply:
I like the sound of that idea.